src/types/block.ts

```diff
@@ -388,19 +387,455 @@ export interface TransactionInBlock {
   /** The transaction as an algosdk `Transaction` object. */
   transaction: Transaction
   /** The asset ID if an asset was created from this transaction. */
-  createdAssetId?: number
+  createdAssetId?: bigint
   /** The app ID if an app was created from this transaction. */
-  createdAppId?: number
+  createdAppId?: bigint
   /** The asset close amount if the sender asset position was closed from this transaction. */
-  assetCloseAmount?: number | bigint
+  assetCloseAmount?: bigint
   /** The ALGO close amount if the sender account was closed from this transaction. */
-  closeAmount?: number
+  closeAmount?: bigint
   /** Any logs that were issued as a result of this transaction. */
   logs?: Uint8Array[]
   /** Rewards in microalgos applied to the close remainder to account. */
-  closeRewards?: number
+  closeRewards?: bigint
   /** Rewards in microalgos applied to the sender account. */
-  senderRewards?: number
+  senderRewards?: bigint
   /** Rewards in microalgos applied to the receiver account. */
-  receiverRewards?: number
+  receiverRewards?: bigint
+}
+
+/**
+ * Interfaces for the encoded transaction object. Every property is labelled with its associated Transaction type property
+ */
+export interface EncodedAssetParams {
```

Comment:
Why do we need all of these types now? Were they previously from algosdk or ...?

Reply:
They used to be in algosdk and exposed via utils-ts. That isn't the case anymore so I had to copy the types from algosdk v2.9 over.